pkg/utils/kubernetes/utils.go

```diff
@@ -0,0 +1,130 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+)
+
+//OnFinalize Callback function to be executed when an object is being deleted
+type OnFinalize func() error
+
+type ObjectFinalizer struct {
+	object     runtime.Object
```

Comment:
Isn't this redundant, once using the owner UUID to store/categorize finalizers, and then again storing an object reference?

Reply:
Yes, this is simplified with the KubernetesResource interface you suggested